#20 Добавлен плагин замены автора коммита

abirambul opened the pull request.

abirambul committed:
#20 Добавлен плагин замены автора коммита

khorevaa commented:
А тесты будут? ))  Или у тебя где то тестировалось?

vkislitsin commented:
Приветствую, коллеги.
 
В коде плагина производится поиск НовогоАвтора среди существующих пользователей Хранилища, причем не всех, а полученных из отчета по версиям, т.е. тех, кто выполнял хотя бы один коммит в Хранилище.
`СтрокаПользователя = ТаблицаПользователейХранилища.Найти(НовыйАвтор, "Автор");` 

В моей практике замена Автора коммита чаще всего требуется для подстановки Пользователя, которого нет в Хранилище.
Например, чтобы авторство для обновлений релизов типовых конфигураций было за пользователем "Компания 1С", а не тем, кто обновлял.

Предложил бы не ограничивать подставляемых Авторов только теми, кто есть в Хранилище и выполнял коммиты в нём.

abirambul committed:
#20 Плагин замены автора коммита

Исправлено поведение плагина.

Для работы плагина требуется только наличие автора в файле AUTHORS. Наличие пользователя в хранилище не обязательно.

abirambul committed:
Merge branch 'develop' into 20-replace-authors

artbear commented:
@khorevaa этот плагин у нас, в команде ИТ-Лаборатории Инфостарта, уже полгода работает активно.

завтра, 19.02, на митапе Инфостарта по CI\CD  автор немного расскажет об этом плагине и покажет примеры его работы.

Алексей, чего не хватает, чтобы пулл-реквест был принят?

khorevaa commented:
Принято! Раз работает то можно и без тестов )
